socorro/scripts/__init__.py

```diff
@@ -75,6 +76,66 @@ def __call__(self, parser, namespace, values, option_string=None):
         setattr(namespace, self.dest, value)
 
 
+class FallbackToPipeAction(argparse.Action):
+    """Fallback to load strings piped from stdin
+
+    This action reads from stdin when the positional arguments were omitted. It
+    expects one value per line, and at least one value.
+
+    This does not work with named arguments, since the action is not called
+    when a named argument is omitted.
+
+    To use this as a fallback for positional arguments:
+    > parser.add_argument('name', nargs='*', action=FallbackToPipeAction)
+    """
+    def __init__(self, option_strings, dest, nargs=None, **kwargs):
+        """Initialize the FallbackToPipeAction.
+
+        :param option_strings: Names for non-positional arguments.
+        :param dest: The destination attribute on the parser
+        :param nargs: The number of arguments, must be '*' for zero or more
+        :param kwargs: Additional parameters for the parser argument
+        """
+        if option_strings:
+            raise ValueError("This action does not work with named arguments")
+        if nargs != '*':
+            raise ValueError("nargs should be '*'")
```

Comment:
Out of curiosity, why restrict it to 0 or more?

Reply:
Great question, and there should be a comment that explains. 

With ``*``, the action is called with an empty list. With ``?`` and some others I tested, it isn't called at all.

Reply:
Oh, how weird. No need for a comment--I was just curious.